File: frontend/src/types/market.ts
export interface IndexQuote {
  name: string
  symbol: string
  price: number
  change: number
  change_percent: number
}

export interface IndexIntradayPoint {
  time: string
  price: number
}

export interface MarketMover {
  ticker: string
  price: number
  change_percent: number
}

export interface MarketMovers {
  gainers: MarketMover[]
  losers: MarketMover[]
}

export interface MarketHeadline {
  title: string
  url: string
  source_name: string
  published_at: string | null
}

File: backend/app/schemas/market.py
from datetime import datetime

from pydantic import BaseModel


class IndexQuoteResponse(BaseModel):
    name: str
    symbol: str
    price: float
    change: float
    change_percent: float


class IndexIntradayPointResponse(BaseModel):
    time: str
    price: float


class MarketMoverResponse(BaseModel):
    ticker: str
    price: float
    change_percent: float


class MarketMoversResponse(BaseModel):
    gainers: list[MarketMoverResponse]
    losers: list[MarketMoverResponse]


class MarketHeadlineResponse(BaseModel):
    title: str
    url: str
    source_name: str
    published_at: datetime | None
